devtools/x/src/utils.rs

```diff
@@ -16,19 +21,102 @@ pub fn project_root() -> &'static Path {
         .unwrap()
 }
 
-pub fn locate_project(xctx: &XContext) -> Result<PathBuf> {
+pub fn locate_project(cargo_config: &CargoConfig) -> Result<PathBuf> {
     #[derive(Deserialize)]
     struct LocateProject {
         root: PathBuf,
     };
 
-    let output = Cargo::new(xctx.config().cargo_config(), "locate-project").run_with_output()?;
+    let output = Cargo::new(cargo_config, "locate-project", false).run_with_output()?;
     Ok(serde_json::from_slice::<LocateProject>(&output)?.root)
 }
 
-pub fn project_is_root(xctx: &XContext) -> Result<bool> {
-    let mut project = locate_project(xctx)?;
+pub fn project_is_root(cargo_config: &CargoConfig) -> Result<bool> {
+    let mut project = locate_project(cargo_config)?;
     project.pop();
 
     Ok(project == project_root())
 }
+
+fn stop_sccache_server() {
+    let mut sccache = Command::new("sccache");
+    sccache.arg("--stop-server");
+    let result = sccache.output();
+    if let Ok(output) = result {
+        if output.status.success() {
+            info!("Stopped already running sccache.");
+        }
+    }
+}
+
+pub fn apply_sccache_if_possible<'a>(
+    cargo_config: &'a CargoConfig,
+) -> Result<Vec<(&'a str, Option<&'a str>)>> {
+    let mut envs = vec![];
+
+    if option_env!("SKIP_SCCACHE").is_none() && cargo_config.sccache.is_some() {
+        if let Some(sccach_config) = &cargo_config.sccache {
```

Comment:
Should the variable name be `sccache_config`?

Reply:
I don't think so, since it can only turn sccache on and off at build time (you were correct to point out option_env! was producing code embedded in the binary.   That was not my intent in the least.